CF Manifest causing server to become unhealthy

From reactor we eventually get error if server is kept running:
```
Caused by: java.io.IOException: Too many open files
	at sun.nio.ch.IOUtil.makePipe(Native Method) ~[na:1.8.0_171]
	at sun.nio.ch.EPollSelectorImpl.<init>(EPollSelectorImpl.java:65) ~[na:1.8.0_171]
	at sun.nio.ch.EPollSelectorProvider.openSelector(EPollSelectorProvider.java:36) ~[na:1.8.0_171]
	at io.netty.channel.nio.NioEventLoop.openSelector(NioEventLoop.java:175) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	... 55 common frames omitted
```

Full trace is:
```
2018-07-23 10:43:27.240  INFO 23096 --- [pool-3-thread-1] .CloudFoundryManifestApplicationDeployer : Checking application status for the release: my-jannepython-printer
2018-07-23 10:43:29.468 ERROR 23096 --- [pool-3-thread-1] .CloudFoundryManifestApplicationDeployer : Caught exception while querying for status of jannepython-printer-v3

java.lang.IllegalStateException: failed to create a child event loop
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:88) ~[netty-common-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58) ~[netty-common-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:47) ~[netty-common-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:59) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:77) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:72) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:59) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at reactor.ipc.netty.resources.DefaultLoopResources.<init>(DefaultLoopResources.java:70) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at reactor.ipc.netty.resources.DefaultLoopResources.<init>(DefaultLoopResources.java:57) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at reactor.ipc.netty.resources.LoopResources.create(LoopResources.java:83) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at org.cloudfoundry.reactor._DefaultConnectionContext.getThreadPool(_DefaultConnectionContext.java:233) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext.access$2601(DefaultConnectionContext.java:25) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext$InitShim.getThreadPool(DefaultConnectionContext.java:215) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext.getThreadPool(DefaultConnectionContext.java:415) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor._DefaultConnectionContext.lambda$getHttpClient$5(_DefaultConnectionContext.java:104) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at reactor.ipc.netty.http.client.HttpClient.<init>(HttpClient.java:135) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at reactor.ipc.netty.http.client.HttpClient.<init>(HttpClient.java:63) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at reactor.ipc.netty.http.client.HttpClient$Builder.build(HttpClient.java:461) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at reactor.ipc.netty.http.client.HttpClient.create(HttpClient.java:86) ~[reactor-netty-0.7.8.RELEASE.jar!/:0.7.8.RELEASE]
	at org.cloudfoundry.reactor._DefaultConnectionContext.getHttpClient(_DefaultConnectionContext.java:101) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext.access$2001(DefaultConnectionContext.java:25) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext$InitShim.getHttpClient(DefaultConnectionContext.java:117) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext.<init>(DefaultConnectionContext.java:76) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext.<init>(DefaultConnectionContext.java:25) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.cloudfoundry.reactor.DefaultConnectionContext$Builder.build(DefaultConnectionContext.java:1041) ~[cloudfoundry-client-reactor-3.12.0.RELEASE.jar!/:na]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.PlatformCloudFoundryOperations.getCloudFoundryOperations(PlatformCloudFoundryOperations.java:49) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.getStatus(CloudFoundryManifestApplicationDeployer.java:133) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryManifestApplicationDeployer.status(CloudFoundryManifestApplicationDeployer.java:180) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryReleaseManager.status(CloudFoundryReleaseManager.java:156) [spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.server.service.ReleaseStateUpdateService.update(ReleaseStateUpdateService.java:93) [spring-cloud-skipper-server-core-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.server.service.ReleaseStateUpdateService$$FastClassBySpringCGLIB$$a59bd08e.invoke(<generated>) [spring-cloud-skipper-server-core-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) [spring-core-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:736) [spring-aop-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) [spring-aop-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99) [spring-tx-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:282) [spring-tx-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) [spring-tx-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) [spring-aop-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:671) [spring-aop-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.cloud.skipper.server.service.ReleaseStateUpdateService$$EnhancerBySpringCGLIB$$8da102ba.update(<generated>) [spring-cloud-skipper-server-core-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at sun.reflect.GeneratedMethodAccessor123.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_171]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_171]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:65) [spring-context-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [na:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [na:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
Caused by: io.netty.channel.ChannelException: failed to open a new selector
	at io.netty.channel.nio.NioEventLoop.openSelector(NioEventLoop.java:177) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoop.<init>(NioEventLoop.java:151) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:127) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:36) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84) ~[netty-common-4.1.25.Final.jar!/:4.1.25.Final]
	... 51 common frames omitted
Caused by: java.io.IOException: Too many open files
	at sun.nio.ch.IOUtil.makePipe(Native Method) ~[na:1.8.0_171]
	at sun.nio.ch.EPollSelectorImpl.<init>(EPollSelectorImpl.java:65) ~[na:1.8.0_171]
	at sun.nio.ch.EPollSelectorProvider.openSelector(EPollSelectorProvider.java:36) ~[na:1.8.0_171]
	at io.netty.channel.nio.NioEventLoop.openSelector(NioEventLoop.java:175) ~[netty-transport-4.1.25.Final.jar!/:4.1.25.Final]
	... 55 common frames omitted
```

--- a/spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/PlatformCloudFoundryOperations.java
+++ b/spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/PlatformCloudFoundryOperations.java
@@ -15,6 +15,9 @@
  */
 package org.springframework.cloud.skipper.deployer.cloudfoundry;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import org.cloudfoundry.client.CloudFoundryClient;
 import org.cloudfoundry.operations.CloudFoundryOperations;
 import org.cloudfoundry.operations.DefaultCloudFoundryOperations;
@@ -33,12 +36,26 @@ public class PlatformCloudFoundryOperations {
 	private static final Logger logger = LoggerFactory.getLogger(PlatformCloudFoundryOperations.class);
 
 	private final CloudFoundryPlatformProperties cloudFoundryPlatformProperties;
+	private final Map<String, CloudFoundryOperations> cache = new HashMap<>();
 
 	public PlatformCloudFoundryOperations(CloudFoundryPlatformProperties cloudFoundryPlatformProperties) {
 		this.cloudFoundryPlatformProperties = cloudFoundryPlatformProperties;
 	}
 
-	public CloudFoundryOperations getCloudFoundryOperations(String platformName) {
+	public synchronized CloudFoundryOperations getCloudFoundryOperations(String platformName) {
+		CloudFoundryOperations operations = cache.get(platformName);
+		if (operations == null) {
+			logger.debug("No existing CloudFoundryOperations for platformName {}, creating new");
+			operations = buildCloudFoundryOperations(platformName);
+			cache.put(platformName, operations);
+		}
+		else {
+			logger.trace("Using existing CloudFoundryOperations for platformName {}");
+		}
+		return operations;
+	}
+
+	private CloudFoundryOperations buildCloudFoundryOperations(String platformName) {
 		CloudFoundryPlatformProperties.CloudFoundryProperties cloudFoundryProperties = this.cloudFoundryPlatformProperties
 																							.getAccounts()
 																							.get(platformName);

